app/serializers/topic_view_serializer.rb

```diff
@@ -239,7 +239,7 @@ def include_pending_posts_count?
   end
 
   def include_tags?
-    SiteSetting.tagging_enabled
+    SiteSetting.tagging_enabled && (!object.topic.private_message? || scope.can_tag_pms?)
```

Comment:
I'm seeing variants of `SiteSetting.tagging_enabled && (!object.topic.private_message? || scope.can_tag_pms?)` being duplicated throughout in this PR. Maybe we can move the logic into a `guardian` method that takes in the topic as a parameter?